mysql/datadog_checks/mysql/execution_plans.py

```diff
@@ -307,32 +320,129 @@ def _collect_execution_plans(self, db, tags, options, min_collection_interval):
         statsd.gauge("dd.mysql.collect_execution_plans.seen_statement_plan_sigs", len(seen_statement_plan_sigs),
                      tags=instance_tags)
 
-    def _run_explain(self, cursor, statement, schema, instance_tags):
-        # TODO: cleaner query cleaning to strip comments, etc.
-        if statement.strip().split(' ', 1)[0].lower() not in VALID_EXPLAIN_STATEMENTS:
-            return
+    def _attempt_explain(self, cursor, statement, schema):
+        """
+        Tries the available methods used to explain a statement for the given schema. If a non-retryable
+        error occurs (such as a permissions error), then statements executed under the schema will be
+        disallowed in future attempts.
+        """
+        explain_strategy_none = 'none'
+        explain_strategy_procedure = 'procedure'
+        explain_strategy_statement = 'statement'
+
+        fns_by_strategy = {
+            explain_strategy_procedure: self._run_explain_procedure,
+            explain_strategy_statement: self._run_explain,
+        }
+
+        plan = None
+        strategy_cache_key = 'explain_strategy:%s' % schema
+        strategy_cache_ttl = 60 * 10
 
+        if not self._can_explain(statement):
+            return None
```

Comment:
May be useful to add a debug log to each of these early returns explaining why we failed to explain the plan. Would help a lot when debugging.